### PR TITLE
Copy files from shared groupfolder in user trash

### LINKED ISSUE #1379: Deleted files not showing in trash
Steps to reproduce:

1. Have a Nextcloud version 20.0.8 or 21.0.0 (have not tested other versions)
2. Install Groupfolder App
3. Create groupfolder with full permissions for a single group (have not tested with multiple groups)
4. Sign in with a user (User1) belonging to the group with permissions for the newly created groupfolder
5. Create a subfolder within the groupfolder and share it with a different user.
6. Sign in with that user (User2)
7. Create a subfolder in the shared folder
8. Share this subfolder by link with only upload permissions
9. Use independent Browser to upload a file via shared link
10. As User2 delete the uploaded file.

Expected behavior:
The deleted file appears in both thrashbins of User1 and User2

Actual behavior:
The deleted file only appears in the trashbin of User1

### PR DESCRIPTION
**Pull request by CarlSchwan**
Needs https://github.com/nextcloud/server/pull/28513 so it's material
for Nextcloud 24

Fix #1379 

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: lib/Trash/TrashBackend.php [hunk 1]
```diff
@@ -170,6 +185,17 @@ public function removeItem(ITrashItem $item) {
 
 	public function moveToTrash(IStorage $storage, string $internalPath): bool {
 		if ($storage->instanceOfStorage(GroupFolderStorage::class) && $storage->isDeletable($internalPath)) {
+			if ($storage->instanceOfStorage(SharedStorage::class)) {
+				// This is a shared group folder so we also do a copy inside the
+				// recipient trash otherwise they can't revert their action.
+				$homeStorage = \OC::$server->get(IRootFolder::class)->getUserFolder($this->userSession->getUser()->getUID())->getStorage();
```
Comment (PVince81) [inline lib/Trash/TrashBackend.php:191]: try injecting the IRootFolder in the constructor

**Review by PVince81 — APPROVED** [review]
:+1: makes sense, please fix the injection though

**Comment by acsfer** [comment]
Some conflicts still 🤔

**Comment by alweber-cap** [comment]
Is this PR still active or ready for merge? We require this functionality in our deployment, but don't want to create a fork just for that.

**Comment by provokateurin** [comment]
Closing as the server PR that seems to be required for it was also closed after long inactivity.

**Pull request closed without merging**